File: PONG.py
import pygame
pygame.init()


screen = pygame.display.set_mode((700,500))
pygame.display.set_caption("PONG")

bx = 350
by = 250
bVx = 5
bVy = 5
p1Score = 0
p2Score = 0

#game loop
doExit = False

#The clock
clock = pygame.time.Clock()


#paddle position
p1x = 20
p1y = 200
p2x = 660
p2y = 200


while not doExit: #GAME LOOP
    
    # event queue
    clock.tick(60)
    
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            doExit = True
            
    #game logic
    keys = pygame.key.get_pressed()
    if keys[pygame.K_w]:
        p1y-=5
    if keys[pygame.K_s]:
        p1y+=5
    if keys[pygame.K_i]:
        p2y-=5
    if keys[pygame.K_k]:
        p2y+=5
    
    #ball movement
    bx += bVx
    by += bVy
    
    #reflect ball off side walls of screen
   
    if by < 0 or by +20 > 500:
        bVy *= -1
            
    #ball-paddle reflection
    if bx < p1x + 20 and by + 20 > p1y and by < p1y + 100:
        bVx *= -1
    if bx + 20 > p2x and by + 20 > p2y and by < p2y + 100:
        bVx *= -1
    
    if bx > 680:
        bVx *= -1
        p1Score += 1
    
    
    if bx < 0:
        bVx *= -1
        p2Score += 1
    
    
    #render section
    screen.fill((0,0,0))
    pygame.draw.rect(screen, (255, 255, 255), (p1x, p1y, 20, 100), 1)
    pygame.draw.rect(screen, (255, 255, 255), (p2x, p2y, 20, 100), 1)
    pygame.draw.rect(screen, (255, 255, 255), (bx, by, 20, 20), 1)
    
    font = pygame.font.Font(None, 74)
    text = font.render(str(p1Score), 1, (255, 255, 255))
    screen.blit(text, (250,10))
    text = font.render(str(p2Score), 1, (255, 255, 255))
    screen.blit(text, (420,10))
    
    # line in the middle
    pygame.draw.line(screen, (255, 255, 255), [349, 0], [349, 500], 5)
   
    
    
    #updates the screen
    pygame.display.flip()
            
            
            
            


#END GAME LOOP
            
pygame.quit()

File: PONG.cpp
// Demonstrate primitve drawing in SFML
#include<iostream>
using namespace std;


#include "SFML/Graphics.hpp"

int main() {

    //game set up (you'll need these two lines in every game)
    sf::RenderWindow renderWindow(sf::VideoMode(500, 500), "Pong Paddle"); //set up screen
    sf::Event event; //set up event queue

    //paddle1 set up
    sf::RectangleShape paddle1(sf::Vector2f(10.0f, 100.0f));
    paddle1.setFillColor(sf::Color::Blue); //other colors: https://www.sfml-dev.org/documentation/2.5.1/classsf_1_1Color.php
    paddle1.setPosition(20.0f, 250.0f); //set position: this is where the top left corner will be


    //paddle2 set up
    sf::RectangleShape paddle2(sf::Vector2f(10.0f, 100.0f));
    paddle2.setFillColor(sf::Color::Blue); //other colors: https://www.sfml-dev.org/documentation/2.5.1/classsf_1_1Color.php
    paddle2.setPosition(470.0f, 250.0f); //set position: this is where the top left corner will be

    //ball set up
    float ballX = 250;
    float ballY = 250;
    float xVel = .1;
    float yVel = .1;
    sf::CircleShape ball(10); //sets radius of circle
    ball.setFillColor(sf::Color(200, 50, 50));//notice you can do numbers here instead of color names
    ball.setPosition(ballX, ballY); //change so it's centered!


    //################### HOLD ONTO YOUR BUTTS, ITS THE GAME LOOP###############################################################
    while (renderWindow.isOpen()) {//keep window open until user shuts it down
        while (renderWindow.pollEvent(event)) { //look for events

            //this checks if the user has clicked the little "x" button in the top right corner
            if (event.type == sf::Event::EventType::Closed)
                renderWindow.close();

            //KEYBOARD INPUT (just one key to start
            if (sf::Keyboard::isKeyPressed(sf::Keyboard::W)) { //checks if "W" is pressed
                paddle1.move(0, -5); //move the rectangle 5 pixels UP on the y axis
                cout << "rectangle moving up" << endl; //for Mo's testing, you can take this out
            }
            if (sf::Keyboard::isKeyPressed(sf::Keyboard::S)) { //checks if "S" is pressed
                paddle1.move(0, 5); //move the rectangle 5 pixels DOWN on the y axis
                cout << "rectangle moving down" << endl; //for Mo's testing, you can take this out
            }

            if (sf::Keyboard::isKeyPressed(sf::Keyboard::I)) { //checks if "W" is pressed
                paddle2.move(0, -5); //move the rectangle 5 pixels UP on the y axis
                cout << "rectangle moving up" << endl; //for Mo's testing, you can take this out
            }
            if (sf::Keyboard::isKeyPressed(sf::Keyboard::K)) { //checks if "S" is pressed
                paddle2.move(0, 5); //move the rectangle 5 pixels DOWN on the y axis
                cout << "rectangle moving down" << endl; //for Mo's testing, you can take this out
            }
        }

        //physics section

        //paddle collision
        if (ballX - 30 < paddle1.getPosition().x && ballY > paddle1.getPosition().y && ballY < paddle1.getPosition().y + 100) {
            cout << "paddle1 collision!" << endl;
            //eventually add sound effect here
            xVel *= -1;

        }

        if (ballX + 30 > paddle2.getPosition().x && ballY > paddle2.getPosition().y && ballY < paddle2.getPosition().y + 100) {
            cout << "paddle2 collision!" << endl;
            //sound effect
            xVel *= -1;
        }
        
        


        //reflects off left and right walls
        if (ballX < 0 || ballX > 500) {
            xVel *= -1;
        }
        if (ballY < 0 || ballY>500) {
            yVel *= -1;
        }

        //add speed to position
        ballX += xVel;
        ballY += yVel;
        //update ball
        ball.setPosition(ballX, ballY);

        //render section-----------------------------------------
        renderWindow.clear(); //wipes screen, without this things smear
        renderWindow.draw(paddle1); //you gotta drew each object
        renderWindow.draw(paddle2);
        renderWindow.draw(ball);
        renderWindow.display(); //flips memory drawings onto screen

    }//######################## end game loop ###################################################################################

    cout << "goodbye!" << endl;
} //end game
